core/da_footprint.go

```
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+const L1AttributesJovianLen = 178
+
+// CalcGasUsedOptimism calculates the gas used for an OP Stack chain.
+// Jovian introduces a DA footprint block limit, which potentially increases the gasUsed.
+// CalcGasUsedOptimism is safe to call on non-Optimism chains or before Jovian.
+func CalcGasUsedOptimism(txs []*types.Transaction, evmGasUsed uint64) uint64 {
+	if len(txs) == 0 || !txs[0].IsDepositTx() || len(txs[0].Data()) < L1AttributesJovianLen {
+		return evmGasUsed
+	}
```

Silent fallbacks like this can be really error prone

---

It's not really meant like a silent fallback, but so that the function can be used pre-Jovian and for non-OP Stack chains, like the docs of the function says. Detecting Jovian being active by the length of the L1Attributes deposit data is exact. The length is 178 iff Jovian is active.

---

But I can make it panic instead and change the docs to require that this function is only called post-Jovian. The two current callers only call this function post-Jovian.

---

I still think it's a little too error-prone, there's no strict type or intent checked, there is a lot about the input implied, and the output isn't obviously wrong when it returns, since `evmGasUsed` can still be a valid value to return in some cases post-jovian. Maybe turn it into a private function, so we at least control the callsites, and maybe add a warning in the doc comment that it doesn't do strict checks?